### DISCUSSION TITLE
Best Way to Convert JPG to PNG Without Quality Loss?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#7727] [Help] [answered]
### ImageMagick version

7.1.0-62-Q16

### Operating system, version and so on

Windows 10

### Description

Hello,

Can anyone confirm the best way to convert jpg to png, if the priority is not to lose any image quality or reduce the amount of quality loss as much as possible. I have gone through the documentation and this is what I understand to be correct:

magick myimage.jpg -quality 100 -define png:compression-level=0 myimage.png

From what I understand, setting the quality to 100 and the compression level to 0 should help keep the image quality intact, but I'd really appreciate any feedback or confirmation from those who have experience with this. Also I have no intent to go back to jpg in after I have converted to png, in case that is relevant.

Thanks in advance for any guidance!

**Comment by fmw42** [comment]
JPG is always lossy even at quality 100.

**Comment by snibgo** [comment, marked as answer]
There is no need to include "`-quality 100 -define png:compression-level=0`". Writing to PNG is always lossless (which mean that no compression method will change data).

**Reply by ghost** [reply]
Thank you. Yes, I did various tests and noticed no difference.
